galley/pkg/config/analysis/analyzers/virtualservice/destinationhosts.go

```diff
@@ -75,13 +78,40 @@ func (d *DestinationHostAnalyzer) analyzeVirtualService(r *resource.Entry, ctx a
 }
 
 func (d *DestinationHostAnalyzer) checkDestinationHost(vsNamespace string, destination *v1alpha3.Destination,
-	ctx analysis.Context, serviceEntryHosts map[resource.Name]bool) bool {
+	ctx analysis.Context, serviceEntryHosts []resource.Name) bool {
 
 	name := util.GetResourceNameFromHost(vsNamespace, destination.GetHost())
-
-	// Check explicitly defined ServiceEntries as well as services discovered from the platform
-	if _, ok := serviceEntryHosts[name]; ok {
-		return true
+	_, cmpHost := name.InterpretAsNamespaceAndName()
+
+	// Check defined ServiceEntries as well as services discovered from the platform
+	for _, seFullHost := range serviceEntryHosts {
+		_, seHost := seFullHost.InterpretAsNamespaceAndName()
+
+		// Check wildcard instances of ServiceEntries
+		if wildcardIndex := strings.Index(seHost, "*"); wildcardIndex > -1 {
+			// Escaping dot chars for RegExp. Dot char means all possible chars.
+			// This protects this validation to false positive for (api-dev.example.com and api.dev.example.com)
+			seHost = strings.Replace(seHost, ".", "\\.", -1)
+
+			if wildcardIndex == 1 {
+				// We need to fix "*" -> ".*" for regexp engine
+				seHost = strings.ToLower(strings.Replace(seHost, "*", ".*", -1))
+			}
+
+			// We anchor the beginning and end of the string when it's
+			// to be used as a regex, so that we don't get spurious
+			// substring matches, e.g., "example.com" matching
+			// "foo.example.com".
+			seHost = strings.Join([]string{"^", seHost, "$"}, "")
+
+			if regexp.MustCompile(seHost).MatchString(cmpHost) {
```

Comment:
Is using regex overkill here? We're not really doing a full regex search, just a wildcard prefix search. I think treating it that way can probably simplify the logic you go through to create a regex string. 

e.g.
```go
	// Wildcard match
	if strings.HasPrefix(seHost, "*") {
		if strings.HasSuffix(cmpHost, strings.TrimPrefix(seHost, "*")) {
			return true
		}
	}
```

Reply:
I guess we need a bit more complex logic than that, from the top of my mind at least the following:

*.host.com -> any.host.com, any.any.host.com (etc), but not host.com
*.host.com <-> * 

And of course exact matches such as * <-> *

But yes, regexp is an overkill.

Reply:
My example should correctly handle the first case (we preserve the dot). 

For the second case, I don't believe the service entry destination allows wildcards. If they do, it isn't documented here: https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/

Reply:
I think it should work if you use ``resolution: NONE`` in the SE? At least issue #7603 discusses this type of configuration.

Reply:
Hooray for undocumented features. :/ 

FYI you're likely to run into merge conflicts with https://github.com/istio/istio/pull/17836